packages/orchestration/src/exos/chain-hub.js

```diff
@@ -679,8 +681,8 @@ export const makeChainHub = (
     },
 
     /**
-     * @param {AccountId | Bech32Address | string} partialId CAIP-10 account ID
-     *   or a Cosmos bech32 address
+     * @param {AccountId | Bech32Address} partialId CAIP-10 account ID or a
+     *   Cosmos bech32 address
```

Comment:
I wonder if it makes sense to leave `| string` here. If we don't, then code that gets addresses from users and wants to attempt the conversion would have to check before calling, or tell typescript to ignore the unknown string.

Actually, since `resolveCosmosChainId` is declared to throw, it should accept strings that are not properly formatted.

Reply:
> I wonder if it makes sense to leave | string here.

I don't think so.

> If we don't, then code that gets addresses from users and wants to attempt the conversion would have to check before calling, or tell typescript to ignore the unknown string.

That is productive. Then the user would know what is accepted. The narrower type makes mistakes like [this](https://github.com/Agoric/agoric-sdk/pull/11174#discussion_r2029575851) less likely.

> since resolveCosmosChainId is declared to throw, it should accept strings that are not properly formatted.

It throws on exceptions. The param type should be unexceptional. For example, it will also throw if the arg is a number, but that's not a reason to include it in the param type.